app/routes/cms.tsx

```diff
@@ -0,0 +1,35 @@
+import { Box, Flex } from "@chakra-ui/react";
+import { json, LoaderFunction, Outlet, useLoaderData } from "remix";
+import { authorize, logout } from "~app/auth";
+import { CmsNavbar } from "~app/nav";
+
+import { PrismaClient as prisma } from "~app/prisma";
+
+export const getLoaderData = async (request: Request) => {
+	const { id } = await authorize(request);
+
+	const user = await prisma.user.findUnique({
+		select: { firstname: true, lastname: true },
+		where: { id },
+	});
+
+	if (!user) throw await logout(request);
+	return user;
+};
+
+type LoaderData = Awaited<ReturnType<typeof getLoaderData>>;
+export const loader: LoaderFunction = async ({ request }) =>
+	json<LoaderData>(await getLoaderData(request));
+
+export default function Index() {
+	const { firstname, lastname } = useLoaderData<LoaderData>();
+
+	return (
+		<Flex w="100%" minH="100vh">
+			<CmsNavbar fullName={`${firstname} ${lastname}`} />
+			<Box flexGrow={1}>
+				<Outlet />
+			</Box>
+		</Flex>
```

Comment:
Why is this being wrapped in a Flex layout, without making use of it?

Reply:
Because the flexbox automaticially aligns these items horizontally, the flexbox is being used. You replace this with a HStack with spacing 0 though

Reply:
Actually using a `HStack` only makes things worse

Reply:
I think the whole Flex wrapping adds additional complexity for child components, which goes against the idea that components should not be aware/care about where they are placed (encapsulation). Why not use conventional methods (`width: 100%`, margin) as in `index.tsx` to horizontally lay-out the page sections?